docs/knowledgebase/learn-substrate/session-keys.md

```diff
@@ -37,24 +54,47 @@ mod app {
 }
 ```
 
-> This code is just an example of the Substrate node at the
+> This code is just an example of a Substrate node at the
 > [time of writing](https://github.com/substrate-developer-hub/substrate-node-template).
 > Refer to the runtime for the most up-to-date implementation.
 
 The default Substrate node implements Session keys in the
 [Session pallet](https://substrate.dev/rustdocs/v3.0.0/pallet_session/).
 
-## Generation and Use
+### Strongly Typed Wrappers
 
-Session keys are hot keys that must be kept online. The individual keys should **not** be used to
+The Session keys from a Substrate node template could use the same cryptography, but serve very different purposes
+in your runtime logic. To prevent the wrong key being used for the wrong operation, strong
+Rust types wrap these keys, keeping them incompatible with one another and ensuring they are only
+used for their intended purpose.
+
+## Generation
+
+As a node operator, you can generate keys using the RPC call
+[`author_rotateKeys`](https://substrate.dev/rustdocs/v3.0.0/sc_rpc/author/trait.AuthorApi.html#tymethod.rotate_keys).
+You will then need to register the new keys on chain using a [`session.setKeys`](https://substrate.dev/rustdocs/v3.0.0/pallet_session/struct.Module.html#method.set_keys) transaction.
+
+> **Note:** for increased security, session keys should be changed every session. This can be done by creating a certificate via signing a session public key
+> and broadcasting it via an extrinsic.
+
+Since session keys are hot keys that must be kept online, the individual keys should **not** be used to
 control funds. All the logic for handling session keys is in the Substrate client, primitives, and
 Session pallet. If one of the Session keys is compromised, the attacker could commit slashable
 behavior.
 
-Node operators can generate keys via the RPC call
-[`author_rotateKeys`](https://substrate.dev/rustdocs/v3.0.0/sc_rpc/author/trait.AuthorApi.html#tymethod.rotate_keys).
-You will then need to register the new keys on chain with a `session.setKeys` transaction.
+### Examples
```

Comment:
Love the cross linking to tutos we have from the KB -more of it please!! 😁 😍

Reply:
Thanks for the review and suggestions! Things like consistency with terminology like WASM is a big one. Thanks for picking those out.

Reply:
But according to [Wasm website](https://webassembly.org/), its official abbreviation is  **Wasm**. I think we should stick to it.

Reply:
Ah my mistake. Agreed. `Wasm` all over from now on. Not sure where I got WASM from 🤷🏼